File: CCC/'21/J3.cpp
#include<bits/stdc++.h>
using namespace std;

int main() {
    while(1){
        int n, last;
        cin >> n;
        if (n==99999)
            return 0;
        if ((n/1000)%10+n/10000==0)
            cout << (last==1?"right":"left") << " " << n%1000 << endl;
        else if (((n/1000)%10+n/10000)%2==0){
            cout << "right " << n%1000 << endl;
            last = 1;
        }
        else{
            cout << "left " << n%1000 << endl;
            last = 0;
        }
    }
    return 0;
}

File: CCC/'21/J3.py
instructions = []
while True:
    instruction = input()
    if instruction == '99999':
        break
    instructions.append(instruction)

prevdir = ''
for i in instructions:
    direction = int(i[:1]) + int(i[1:2])
    if direction == 0:
        direct = prevdir
    elif direction%2 == 0:
        direct = 'right'
    else:
        direct = 'left'
    prevdir = direct


    print(f'{prevdir} {i[2:]}')